### PR TITLE
Update APIClient interfaces - Closes #5922

### LINKED ISSUE #5922: Add basic scaffold and types for APIClient
### Description

Update index.ts to export
```
  createIPCClient(socketPath: string): Promise<APIClient>;
  createWSClient(url: string): Promise<APIClient>;
  createClient(channel: Channel): Promise<APIClient>;
```
and create empty `APIClient` with unit test sample.
Add `schema` and `nodeInfo` variable in APIClient

### Motivation

Scaffold needs to be created to add name spaces and connection development can be done in parallel. 

### Acceptance Criteria

- Add file structure and sample tests

### Additional Information
Related issue #5919

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?

This PR resolves #5922 

### How was it solved?

- Remove all outdated files
- Add scaffoldings for APIClient, create client functions, and node.ts as a sample for node namespace
- Add simple unit tests for API client

### How was it tested?

- Add unit test

### CODE REVIEW

File: elements/lisk-api-client/src/api_client.ts [hunk 1]
```diff
@@ -12,163 +12,44 @@
...
-			genesisBlockTransactionRoot: constants.TESTNET_NETHASH,
-			...options,
-		});
+	public constructor(channel: Channel) {
+		this._channel = channel;
 	}
 
-	public banActiveNode(): boolean {
-		return this.banNode(this.currentNode);
+	public async connect(): Promise<void> {
```
Comment (pablitovicente) [inline elements/lisk-api-client/src/api_client.ts:28]: Will this method carry over extra tasks once more components of the client are implemented? 
Asking right now it's only invoking some actions and instantiating nodes and the name `connect()` usually implies opening some sort of communication channel.

File: elements/lisk-api-client/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+export type EventCallback<T> = (event: EventInfoObject<T>) => void | Promise<void>;
+
+export interface Channel {
+	connect: () => Promise<void>;
+	disconnect: () => Promise<void>;
+	invoke: <T>(actionName: string, params?: Record<string, unknown>) => Promise<T>;
+	subscribe: <T>(eventName: string, cb: EventCallback<T>) => void;
+}
+
+export interface RegisteredSchema {
```
Comment (pablitovicente) [inline elements/lisk-api-client/src/types.ts:32]: Maybe rename to `RegisteredSchemas` (plural as it seems to hold all the schemas that exist)

**Review by pablitovicente — APPROVED** [review]

File: elements/lisk-api-client/src/api_client.ts [hunk 3]
```diff
@@ -12,163 +12,44 @@
...
+		this._schemas = await this._channel.invoke<RegisteredSchema>('app:getSchema');
+		this._nodeInfo = await this._channel.invoke<NodeInfo>('app:getNodeInfo');
+		this._node = new Node(this._channel);
```
Comment (ManuGowda) [inline elements/lisk-api-client/src/api_client.ts:31]:
```suggestion
              this._node = new Node(this._channel);
		this._nodeInfo = await this._node.info();
```

File: elements/lisk-api-client/src/api_client.ts (hunk 1 above)
Comment (shuse2) [reply elements/lisk-api-client/src/api_client.ts:28]: It was calling `channel.connect` in the draft, but actually if we accept opened channel as input for `createClient` it cannot call `connect` internally anymore.
I will update to `init`

File: elements/lisk-api-client/test/api_client.spec.ts [hunk 4]
```diff
@@ -12,346 +12,65 @@
...
-	describe('#isBanned', () => {
-		it('should return true when provided node is banned', () => {
-			apiClient.bannedNodes = [...apiClient.bannedNodes, localNode];
-			return expect(apiClient.isBanned(localNode)).toBe(true);
-		});
-
-		it('should return false when provided node is not banned', () => {
-			return expect(apiClient.isBanned(localNode)).toBe(false);
+	describe('when invoke is called', () => {
+		it('should disconnect from the channel', async () => {
```
Comment (ManuGowda) [inline elements/lisk-api-client/test/api_client.spec.ts:58]: i did not understand the statement, when invoked why it should disconnect?

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**